maltd-frontend/src/components/hoc/AuthenticationGuard.js

```diff
@@ -46,8 +50,15 @@ export default function AuthenticationGuard() {
       })
       .success(() => {
         keycloak.loadUserInfo().success();
-        localStorage.setItem("jwt", keycloak.token);
-        setAuthedKeycloak(keycloak);
+        if(accessRole != null && accessRole != "" && keycloak.tokenParsed.realm_access.roles.indexOf(accessRole) != -1){
+            localStorage.setItem("jwt", keycloak.token);
+            setAuthedKeycloak(keycloak);
+        }else{
+             keycloak.clearToken();
+             localStorage.clear();
+             alert('Authenticated but not Authorized, request access from your portal administrator');                 
```

Comment:
Question: Is an alert pop up the intended behaviour here?

Reply:
I'm not a React developer haha, was hoping you caught this with a suggestion of how to cleanly inform the user without antiquated methods like these